src/api/transaction.go

```diff
@@ -259,3 +260,75 @@ func getRawTx(gateway Gatewayer) http.HandlerFunc {
 		return
 	}
 }
+
+// VerifyTxnRequest represents the data struct of the request for /transaction/verify
+type VerifyTxnRequest struct {
+	EncodedTransaction string `json:"encoded_transaction"`
+}
+
+func verifyTxHandler(gateway Gatewayer) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodPost {
+			wh.Error405(w)
+			return
+		}
+
+		if r.Header.Get("Content-Type") != "application/json" {
+			wh.Error415(w)
+			return
+		}
+
+		var req VerifyTxnRequest
+		if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+			wh.Error400(w, err.Error())
+			return
+		}
+
+		tx, err := decodeTxn(req.EncodedTransaction)
+		if err != nil {
+			wh.Error422(w, err.Error())
+			return
+		}
+
+		inputs, err := gateway.VerifyTxnVerbose(tx)
+		if err != nil {
+			wh.Error422(w, err.Error())
+			return
+		}
+
+		txRsp, err := NewCreatedTransaction(tx, inputs)
+		if err != nil {
+			wh.Error400(w, err.Error())
+			return
+		}
+
+		rsp := struct {
```

Comment:
Should be outside of the function, also should not use `interface{}` if not necessary